aws-lambda/src/databricks_cdk/resources/service_principals/service_principal_secrets.py

```diff
@@ -0,0 +1,156 @@
+class ServicePrincipalSecretsCreationError(Exception):
+    pass
+
+
+class ServicePrincipalSecretsProperties(BaseModel):
+    service_principal_id: int
+
+
+class ServicePrincipalSecretsResponse(CnfResponse):
+    secrets_manager_arn: str
+    secrets_manager_name: str
+
+
+def create_or_update_service_principal_secrets(
+    properties: ServicePrincipalSecretsProperties, physical_resource_id: Optional[str] = None
+) -> ServicePrincipalSecretsResponse:
+    """
+    Create or update service principal secrets on databricks.
+    If service principal secrets already exist, it will return the existing service principal secrets.
+    If service principal secrets doesn't exist, it will create a new one.
+    """
+    account_client = get_account_client()
+
+    if physical_resource_id:
+        return update_service_principal_secrets(properties, physical_resource_id, account_client)
+
+    return create_service_principal_secrets(properties, account_client)
+
+
+def get_service_principal_secrets(
+    service_principal_id: int, physical_resource_id: str, account_client: AccountClient
+) -> SecretInfo:
+    """Get service principal secrets on databricks based on physical resource id and service principal id."""
+    existing_service_principal_secrets = account_client.service_principal_secrets.list(
+        service_principal_id=service_principal_id
+    )
+
+    for secret_info in existing_service_principal_secrets:
+        if secret_info is not None and secret_info.id == physical_resource_id:
+            return secret_info
+    else:
+        raise NotFound(f"Service principal secrets with id {physical_resource_id} not found")
+
+
+def create_service_principal_secrets(
+    properties: ServicePrincipalSecretsProperties, account_client: AccountClient
+) -> ServicePrincipalSecretsResponse:
+    """
+    Create service principal secrets on databricks.
+    It will create a new service principal secrets and store it in secrets manager.
+    """
+    service_principal = get_service_principal(properties.service_principal_id, account_client)
+    created_service_principal_secrets = account_client.service_principal_secrets.create(
+        service_principal_id=properties.service_principal_id
+    )
+
+    if created_service_principal_secrets.id is None:
+        raise ServicePrincipalSecretsCreationError("Failed to create service principal secrets")
+
+    secret_name = f"{SECRETS_MANAGER_RESOURCE_PREFIX}/{service_principal.display_name}/{service_principal.id}"
+    secrets_manager_resource = add_to_secrets_manager(
+        secret_name=secret_name,
+        client_id=service_principal.application_id,
+        client_secret=created_service_principal_secrets.secret,
+    )
+    return ServicePrincipalSecretsResponse(
+        physical_resource_id=created_service_principal_secrets.id,
+        secrets_manager_arn=secrets_manager_resource["ARN"],
+        secrets_manager_name=secrets_manager_resource["Name"],
+    )
+
+
+def update_service_principal_secrets(
```

Comment:
I would change the function name here, because we are not updating. Maybe `get_service_principal_secrets`?

Reply:
Fixed, https://github.com/godatadriven/databricks-cdk/pull/1074/commits/0680adac7089e6a4c424d06833a72a3e4c1176ae